server/local-tiler.js

```diff
@@ -0,0 +1,169 @@
+"use strict";
+
+var fs = require("fs"),
+    os = require("os"),
+    path = require("path"),
+    url = require("url"),
+    util = require("util");
+
+var async = require("async"),
+    decider = require("swfr").decider,
+    Promise = require("bluebird"),
+    range = require("range").range,
+    rimraf = require("rimraf"),
+    mercator = new (require("sphericalmercator"))(),
+    tmp = require("tmp");
+
+var tile = require("./lib/tile-cells");
+
+Promise.promisifyAll(fs);
+Promise.promisifyAll(tmp);
+rimraf = Promise.promisify(rimraf);
+
+var CELL_HEIGHT = 256,
+    CELL_WIDTH = CELL_HEIGHT;
+
+var worker = decider({
+  sync: true
+}, function(chain, input) {
+  return chain
+    .then(function() {
+      // Keep the directory, we will clean it up manually.
+      var tmpOptions = { };//unsafeCleanup : true };
+      if (input.tmpDir) { 
+        tmpOptions.dir = input.tmpDir; 
+      }
+
+      return tmp.dirAsync(tmpOptions);
+    })
+    .then(function(tmpObj) {
```

Comment:
[`spread`](https://github.com/petkaantonov/bluebird/blob/master/API.md#spreadfunction-fulfilledhandler--function-rejectedhandler----promise) might make this slightly more readable.